packages/cdk8s-plus-17/src/service.ts

```diff
@@ -232,6 +244,10 @@ export class Service extends Resource {
       throw new Error('A service must be configured with a port');
     }
 
+    if (this.type === ServiceType.EXTERNAL_NAME && this.externalName === undefined) {
+      throw new Error('A service with type EXTERNAL_NAME requires an externalName prop');
```

Comment:
Should we also check the opposite? I.e if an externalName is specified then the type must either be undefined or SERVICE_NAME?

Reply:
Added the inverse check in as well